aptos-move/framework/aptos-token/sources/token.move

```diff
@@ -711,7 +711,41 @@ module aptos_token::token {
     }
 
     //
-    // Functions for mutating tokendata fields
+    // Functions for mutating CollectionData fields
+    //
+
+    fun assert_collection_exists(creator_address: address, collection_name: String) acquires Collections {
+        assert!(exists<Collections>(creator_address), error::not_found(ECOLLECTIONS_NOT_PUBLISHED));
+        let all_collection_data = &borrow_global<Collections>(creator_address).collection_data;
+        assert!(table::contains(all_collection_data, collection_name), error::not_found(ECOLLECTION_NOT_PUBLISHED));
+    }
+
+    public fun mutate_collection_description(creator: &signer, collection_name: String, description: String) acquires Collections {
+        let creator_address = signer::address_of(creator);
+        assert_collection_exists(creator_address, collection_name);
+        let collection_data = table::borrow_mut(&mut borrow_global_mut<Collections>(creator_address).collection_data, collection_name);
+        assert!(collection_data.mutability_config.description, error::permission_denied(EFIELD_NOT_MUTABLE));
+        collection_data.description = description;
+    }
+
+    public fun mutate_collection_uri(creator: &signer, collection_name: String, uri: String) acquires Collections {
+        let creator_address = signer::address_of(creator);
+        assert_collection_exists(creator_address, collection_name);
+        let collection_data = table::borrow_mut(&mut borrow_global_mut<Collections>(creator_address).collection_data, collection_name);
+        assert!(collection_data.mutability_config.uri, error::permission_denied(EFIELD_NOT_MUTABLE));
+        collection_data.uri = uri;
+    }
+
+    public fun mutate_collection_maximum(creator: &signer, collection_name: String, maximum: u64) acquires Collections {
```

Comment:
I think we need to check the new maximum is bigger than supply. Otherwise, lead to strange state.

Reply:
good point just updated! can maximum be equal to the supply?

Reply:
Yes, could be equal. Just realized we probably need to do similar check when mutating tokendata maximum

Reply:
What happens if they change the maximum to 0? In general that means that supply should not be tracked. But what if that's tracked before? Similar question for changing maximum from 0 (supply not tracked so many tokens could have been minted) to non-zero

Reply:
ok just updated!

Reply:
we are disabling going from tracked to untracked and vice versa for now